.github/actions/build-rpms/main.js

```diff
@@ -0,0 +1,37 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+const child_process = require("child_process");
+const spawnArgs = {
+	stdio: "inherit",
+	stderr: "inherit",
+};
+
+let atcComponent = process.env.ATC_COMPONENT;
+const dockerComposeArgs = ["-f", `${process.env.GITHUB_WORKSPACE}/infrastructure/docker/build/docker-compose.yml`, "run", "--rm"];
+if (typeof atcComponent !== "string" || atcComponent.length === 0) {
+	console.error("Missing environment variable ATC_COMPONENT");
+	process.exit(1);
+}
+const nonRpmComponents = ["source", "weasel", "docs"];
```

Comment:
I don't think there's a need to support these.

GHA will always run with a specific git ref, so if you want the source it should be as easy as `git checkout {{ref}}`. 

Weasel is already covered by our weasel action - its output could be added as an artifact, if desired, but otherwise the text log is already there, even in raw form for direct linking.

The docs would be better handled by [a dedicated docs workflow](https://github.com/ocket8888/trafficcontrol/blob/master/.github/workflows/docs.yml) which not only checks if they build, but also highlights errors and warnings inline for developers. To output docs artifacts in arbitrary format, see [the `sphin-action` action's README section on the topic of pairing with an artifact uploading action](https://github.com/ammaraskar/sphinx-action#great-actions-to-pair-with)

Reply:
The workflow wasn't running Weasel, but removing that check for `_build` sounds fine to me. Let's address docs in a different PR. Removed docs and source tarball from the workflow in 9398ab696.

Reply:
well the weasel build just runs weasel in a container and puts its output in `dist/weasel.log`